node/consensus/src/lib.rs

```diff
@@ -383,7 +382,7 @@ impl<C, A> bft::Proposer<<C as AuthoringApi>::Block> for Proposer<C, A> where
 			&inherent
 		) {
 			Ok(Ok(())) => None,
-			Ok(Err(BlockBuilderError::TimestampInFuture(timestamp))) => Some(timestamp),
+			Ok(Err(InherentError::Timestamp(TimestampInherentError::TimestampInFuture(timestamp)))) => Some(timestamp),
```

Comment:
this file/module will now be moved to core and as such this module should work even when timestamp isn't part of the runtime. so you can't use `InherentError::Timestamp` or `TimestampInherentError::TimestampInFuture`. the first approach with `BBError` was fine - allow for a generic "please wait X seconds" error - don't tie it to a specific module.